tests/scanner/scanners/bigquery_scanner_test.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2018 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for BigqueryScanner."""
+
+import collections
+import json
+import unittest
+import mock
+
+from tests.scanner.test_data import fake_bigquery_scanner_data as fbsd
+from tests.unittest_utils import ForsetiTestCase
+from google.cloud.forseti.common.gcp_type import resource as resource_mod
+from google.cloud.forseti.scanner.scanners import bigquery_scanner
+
+
+def _mock_gcp_resource_iter(_, resource_type):
+    """Creates a list of GCP resource mocks retrieved by the scanner."""
+    resources = []
+    if resource_type != 'dataset_policy':
+        raise ValueError(
+            'unexpected resource type: got %s, want dataset_policy',
+            resource_type,
+        )
+
+    Resource = collections.namedtuple(
+        'Resource',
+        # fields based on required fields from Resource in dao.py.
+        ['full_name', 'type', 'name', 'parent_type_name', 'parent',
+         'data'],
+    )
+
+    for resource in fbsd.BIGQUERY_DATA:
+        proj = Resource(
+            full_name=resource['full_project_name'],
+            type='project',
+            name='projects/' + resource['project_id'],
+            parent_type_name='',
+            parent=None,
+            data='',
+        )
+
+        dataset= Resource(
+            full_name=resource['full_name'],
```

Comment:
This full_name is the name of the dataset_policy. Trim the end off: '/'.join(full_name.split('/')[:-3]) + '/'

Reply:
Done.